src/vmware/azext_vmware/_validators.py

```diff
@@ -19,3 +21,9 @@ def example_name_or_id_validator(cmd, namespace):
                 type='storageAccounts',
                 name=namespace.storage_account
             )
+
+
+def server_addresses_length(namespace):
+    if isinstance(namespace.server_addresses, list):
+        if len(namespace.server_addresses) > 3:
+            raise CLIError("Enter a maximum of 3 DHCP relay addresses")

```

Comment:
You have to use a more [specific error](ttps://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md#error-type).

Reply:
https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md#error-type